File: test/fixtures/r-httr2/put_basic_auth_json_data.r
library(httr2)

request("http://localhost:28139/test/_security") |>
  req_method("PUT") |>
  req_body_raw(
    '{"admins":{"names":[], "roles":[]}, "readers":{"names":["joe"],"roles":[]}}',
    type = "application/x-www-form-urlencoded"
  ) |>
  req_auth_basic("admin", "123") |>
  req_perform()

File: test/fixtures/java-unirest/put_basic_auth_json_data.java
import kong.unirest.HttpResponse;
import kong.unirest.Unirest;

class Main {
    public static void main(String[] args) {
            HttpResponse<String> httpResponse
               = Unirest.put("http://localhost:28139/test/_security")
            .basicAuth("admin", "123")
            .header("Content-Type", "application/x-www-form-urlencoded")
            .body("{\"admins\":{\"names\":[], \"roles\":[]}, \"readers\":{\"names\":[\"joe\"],\"roles\":[]}}")
            .asString();
    }
}
